src/core/http/mod.rs

```diff
@@ -24,3 +24,10 @@ mod telemetry;
 
 pub static TAILCALL_HTTPS_ORIGIN: HeaderValue = HeaderValue::from_static("https://tailcall.run");
 pub static TAILCALL_HTTP_ORIGIN: HeaderValue = HeaderValue::from_static("http://tailcall.run");
+
+#[derive(Default, Clone, Debug)]
+/// User can configure the filter/interceptor
+/// for the http requests.
+pub struct HttpFilter {
+    pub on_request: Option<String>,
```

Comment:
I think `on_request` should not be optional. There is no utility a HttpFilter that doesn't have any on_request.